tests/findex.test.ts

```diff
@@ -462,3 +468,175 @@ test("upsert and search memory", async () => {
+    expect(indexedValues.length).toEqual(30)
+
+    const locations = results.get("France")
+    expect(locations.length).toEqual(30)
+  }
+})
+
+/**
+ * @param dbFilepath path of sqlite database
+ * @returns nothing
+ */
+async function verify(dbFilepath: string): Promise<void> {
+  const findex = await Findex()
+  const masterKey = new FindexKey(
+    Uint8Array.from(Buffer.from(FINDEX_TEST_KEY, "base64")),
+  )
+  const label = new Label(FINDEX_TEST_LABEL)
+  const db = new Database(dbFilepath)
+  const callbacks = callbacksExamplesBetterSqlite3(db, false)
+
+  //
+  // Verifying search results
+  //
+  {
+    const results = await findex.search(
+      ["France"],
+      masterKey,
+      label,
+      callbacks.fetchEntries,
+      callbacks.fetchChains,
+    )
+
+    const indexedValues = results.getAllIndexedValues("France")
+    expect(indexedValues.length).toEqual(30)
+
+    const locations = results.get("France")
+    expect(locations.length).toEqual(30)
+  }
+
+  //
+  // Upsert a single user
+  //
+  {
+    const newIndexedEntries: IndexedEntry[] = []
+    const newUser = {
+      id: 10000,
+      firstName: "another first name",
+      lastName: "another last name",
+      phone: "another phone",
+      email: "another email",
+      country: "France",
+      region: "another region",
+      employeeNumber: "another employee number",
+      security: "confidential",
+    }
+    newIndexedEntries.push({
+      indexedValue: new Location(toBeBytes(newUser.id)),
+      keywords: [
+        newUser.firstName,
+        newUser.lastName,
+        newUser.phone,
+        newUser.email,
+        newUser.country,
+        newUser.region,
+        newUser.employeeNumber,
+        newUser.security,
+      ],
+    })
+
+    await findex.upsert(
+      newIndexedEntries,
+      masterKey,
+      label,
+      callbacks.fetchEntries,
+      callbacks.upsertEntries,
+      callbacks.insertChains,
+    )
+  }
+
+  //
+  // Another search
+  //
+  {
+    const results = await findex.search(
+      ["France"],
+      masterKey,
+      label,
+      callbacks.fetchEntries,
+      callbacks.fetchChains,
+    )
+
+    const indexedValues = results.getAllIndexedValues("France")
+    expect(indexedValues.length).toEqual(31)
+
+    const locations = results.get("France")
+    expect(locations.length).toEqual(31)
+  }
+}
+
+/**
+ * @param directory directory to list files
+ * @returns array of paths
+ */
+async function readDirectory(directory: string): Promise<string[]> {
+  const files = await fs.promises.readdir(directory)
+  return files
+}
+
+test("Verify Findex non-regression test", async () => {
+  const testFolder = "tests/data/findex/non_regression/"
+  const files = await readDirectory(testFolder)
+  for (const file of files) {
+    const testFilepath = testFolder + file
+    const newFilepath = os.tmpdir() + "/" + file
+    fs.copyFileSync(testFilepath, newFilepath)
+    await verify(newFilepath)
+    console.log("... OK: " + newFilepath)
```

Comment:
Don't think it's necessary to log here, if there is an error it will be reported to the console.

Reply:
in the CI:
- the first time the tests run, only 1 database is used.
- the second time, multiple files must be verified

Reply:
removed